### PR TITLE
fix: --spec now allows () in filename

### LINKED ISSUE #28509: Cypress doesn't like parenthesis () in filename
### Current behavior

  > E:\Work_galaktika\TMS.Agent.Cypress\tms.agent.cypress\cypress\e2e\Наличие в поисковиках\YandexSearch_CorpGal().cy.js
PS E:\Work_galaktika\TMS.Agent.Cypress\tms.agent.cypress> npx cypress run --spec "cypress/e2e/Наличие в поисковиках/YandexSearch_CorpGal().cy.js"

DevTools listening on ws://127.0.0.1:4846/devtools/browser/c773b362-8b8a-40e9-8f8a-5e06968b6350
Can't run because no spec files were found.

We searched for specs matching this glob pattern:
 > E:\Work_galaktika\TMS.Agent.Cypress\tms.agent.cypress\cypress\e2e\Наличие в поисковиках\YandexSearch_CorpGal().cy.js

[Cypress_Debug_log_error.txt](https://github.com/cypress-io/cypress/files/13649432/Cypress_Debug_log_error.txt)

### Desired behavior

_No response_

### Test code to reproduce

test file with parenthesis () in filename
execute command  npx cypress run --spec  "cypress/e2e/filename"

### Cypress Version

13.5.1

### Node version

18.16.1

### Operating System

windows 11

### Debug Logs

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by TomasTeixeira2003**
- Closes https://github.com/cypress-io/cypress/issues/28509

### Additional details
Even though there was an existing workaround, it is easier to just allow parenthesis.

### PR Tasks

- [x] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,9 @@ _Released 4/16/2024 (PENDING)_
 
 - Updated the Chrome flags to not show the "Enhanced Ad Privacy" dialog. Addresses [#29199](https://github.com/cypress-io/cypress/issues/29199).
 
+**Bugfixes**
+ - Specs with () in the filename will no longer fail to load and now behave as any other spec. Fixes [#28509](https://github.com/cypress-io/cypress/issues/28509)
```
Comment (jennifer-shehane) [inline cli/CHANGELOG.md]:
```suggestion
**Bugfixes**

 - Specs with () in the filename will no longer fail to load and now behave as any other spec. Fixes [#28509](https://github.com/cypress-io/cypress/issues/28509)
```

**Comment by TomasTeixeira2003** [comment]
Is there anything I should do since the check for ''Semantic Pull Request'' is failing?

File: cli/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ _Released 4/16/2024 (PENDING)_
 
 - Updated the Chrome flags to not show the "Enhanced Ad Privacy" dialog. Addresses [#29199](https://github.com/cypress-io/cypress/issues/29199).
 
+**Bugfixes**
```
Comment (jennifer-shehane) [inline cli/CHANGELOG.md]: @TomasTeixeira2003 Seems to be just missing a colon here. 
```suggestion
**Bugfixes:**
```

**Review by jennifer-shehane — APPROVED** [review]
I tested this manually and confirmed that the tests now run when specifying a spec with opening and closing parens. 

#### Before

Screenshot 2024-04-11 at 1 53 41 PM




#### After

Screenshot 2024-04-11 at 1 50 03 PM

File: cli/CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,10 @@ _Released 4/11/2024_
 - Suppresses benign warnings that reference Vulkan on GPU-less hosts. Addresses [#29085](https://github.com/cypress-io/cypress/issues/29085). Addressed in [#29278](https://github.com/cypress-io/cypress/pull/29278).
 
 
+**Bugfixes:**
+
+ - Specs with () in the filename will no longer fail to load and now behave as any other spec. Fixes [#28509](https://github.com/cypress-io/cypress/issues/28509)
```
Comment (AtofStryker) [inline cli/CHANGELOG.md]:
```suggestion
 - Specs with () in the filename will no longer fail to load and now behave as any other spec when run with the `--spec` argument via `cypress run`. Fixes [#28509](https://github.com/cypress-io/cypress/issues/28509)
```

Comment (AtofStryker) [reply cli/CHANGELOG.md]: We also need to move this to a new change entry since 13.7.3 was released yesterday

**Review by AtofStryker — COMMENTED** [review]
I don't know if we want to escape parenthesis in all cases in cypress. the `--spec` arg can support glob patterns and parenthesis are a valid glob pattern, but I haven't been able to get glob patterns outside of wildcards to work correctly. @jennifer-shehane do you have more knowledge on what's expected? I mainly going off of https://docs.cypress.io/guides/guides/command-line#cypress-run-spec-lt-spec-gt and testing with the CLI and using https://globster.xyz/

**Comment by jennifer-shehane** [comment]
@AtofStryker That's a good call out. This bug only happens when you have a closing paren DIRECTLY following an opening paren, so `()` not `(a)`. I don't think that would be a valid usecase in any globs 🤔 , so escaping them back to back should be fine. That's what I'm thinking.

**Review by AtofStryker — APPROVED** [review]

**Comment by AtofStryker** [comment]
@jennifer-shehane is the failure in CI legit in relation to the glob changes?

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
@TomasTeixeira2003 There does appear to be a snapshot that has changed with globs - that is failing here. Can you take a look: https://app.circleci.com/pipelines/github/cypress-io/cypress/60955/workflows/e499d77f-ccf3-4aba-8f7f-ed00d2595ba1/jobs/2535654

Screenshot 2024-04-16 at 2 19 19 PM

**Comment by TomasTeixeira2003** [comment]
I will look into it today

**Comment by jennifer-shehane** [comment]
@tomasbjerre Any chance to look at this?

**Comment by tomasbjerre** [comment]
> @tomasbjerre Any chance to look at this?

@jennifer-shehane you probably tagged the wrong Tomas. But I think it looks great :)

**Comment by jennifer-shehane** [comment]
Sorry @TomasTeixeira2003 Any chance to look at this?

**Comment by TomasTeixeira2003** [comment]
I'm sorry, have had a lot of work lately. By tomorrow I will check it.

**Comment by jennifer-shehane** [comment]
@TomasTeixeira2003 Just pinging to check in on these updates again.

**Comment by jennifer-shehane** [comment]
@TomasTeixeira2003 Any chance you'll be able to fix this PR? We'll need to close otherwise.

**Comment by TomasTeixeira2003** [comment]
I've been working on another feature to rerun failed tests and didn't remember about this. Sorry. By Sunday i will have it working.

**Comment by TomasTeixeira2003** [comment]
What is the yarn command to run only the system-tests?

**Comment by jennifer-shehane** [comment]
@TomasTeixeira2003 The instructions are here: https://github.com/cypress-io/cypress/blob/develop/system-tests/README.md

**Comment by jennifer-shehane** [comment]
Rerunning the tests here

**Comment by jennifer-shehane** [comment]
@TomasTeixeira2003 There are some linting errors

**Comment by TomasTeixeira2003** [comment]
And the other ones are flaky?

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
@TomasTeixeira2003 I'm not seeing this fix working anymore when passing the `--spec` arg. I understand this may be isolated to running Cypress in the local mode, but I can't see the change fixing the issue now with the recent changes. Including a log of what I was running. The test.cy.js file ran, but the test().cy.js file did not run.

Screenshot 2024-06-24 at 9 53 01 AM

**Comment by TomasTeixeira2003** [comment]
@jennifer-shehane if I pass the spec file without any blackslash before the parenthesis, bash doesn't even accept the command. But if i do it with the backslash it does work.
Screenshot from 2024-07-12 13-42-07
Screenshot from 2024-07-12 13-42-25

**Comment by jennifer-shehane** [comment]
@TomasTeixeira2003 Are you planning to investigate further?

**Comment by TomasTeixeira2003** [comment]
I don't know what more i can do since my machine allows me to run the spec this way.

**Pull request closed without merging**